### PR TITLE
Hbv4 ib test

### PR DESCRIPTION
**Pull request by rafsalas19**
- Modify perf test build to make compatible for CPU + GPU
- Custom setup script now checks if GPUs are present and builds tests accordingly.
- Add IB write test for HBv4

### CODE REVIEW

File: customTests/azure_ib_write_bw.nhc [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+IB_WRITE_BW_EXE_PATH=/opt/azurehpc/test/nhc/perftest-4.5/ib_write_bw
```
Comment (moes1) [inline customTests/azure_ib_write_bw.nhc:3]: Why do we need to use this ib_write_bw? Is there a problem with the one installed at /usr/bin/?

File: customTests/azure_ib_write_bw.nhc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+IB_WRITE_BW_ARGS="-s $(( 1 * 1024 * 1024 )) -D ${IB_WRITE_BW_DURATION} -x 0 -F --report_gbits -d mlx5_ib0"
+SLEEP_TIME=5
+
+HOSTNAME=`hostname`
+
+function check_ib_bw(){
+
+EXP_IB_BW=$1
+
+IB_WRITE_BW_OUT1=$( taskset -c 0 $IB_WRITE_BW_EXE_PATH $IB_WRITE_BW_ARGS  > /dev/null &)
```
Comment (moes1) [inline customTests/azure_ib_write_bw.nhc]: is taskset better than using numactl? With numactl you can add --membind, although I'm not sure how important that is for this test.

File: customTests/azure_ib_write_bw.nhc (hunk 1 above)
Comment (rafsalas19) [reply customTests/azure_ib_write_bw.nhc:3]: So, the reasoning is that I am not sure it is built with CUDA or not.  I build from source wrt to whether or not the VM has a GPU or not. I think Jithin mentioned that we should work the correct build method onto our images. Also, we want to be able to run Health checks on non-hpc/AI images as well. They may not have that tool installed.

File: customTests/azure_ib_write_bw.nhc (hunk 2 above)
Comment (rafsalas19) [reply customTests/azure_ib_write_bw.nhc]: let me test this out to see if I see a difference. If not I am inclined to leave it alone. ALso, I think Numa mapping becomes more important when there is more than one IB nic. These tests are for one HB series which only have one IB Nic.

Comment (rafsalas19) [reply customTests/azure_ib_write_bw.nhc]: I changed it to numactl --membind=0 and it's working as intended

Comment (rafsalas19) [reply customTests/azure_ib_write_bw.nhc]: Fixed per off line discussion

**Review by moes1 — APPROVED** [review]

**Pull request merged**